thorlcr/slave/slave.cpp

```diff
@@ -287,6 +289,18 @@ class CGenericSlaveGraphElement : public CSlaveGraphElement
         }
         haveCreateCtx = true;
     }
+    virtual CActivityBase *queryActivity()
+    {
+        if (isEof)
```

Comment:
'isEof' is solely used for this purposed, it isn't used (as the name might suggest) as a end-of-input marker.
i.e. I could search and replace it to isNullInput of suchlike and it would have no effect on any other mechanism.

But name is confusing, I'm going to change to 'hasNullInput'